fdbclient/NativeAPI.actor.cpp

```diff
@@ -3025,6 +3027,7 @@ ACTOR Future<Standalone<RangeResultRef>> getRange(Database cx,
 						throw deterministicRandom()->randomChoice(
 						    std::vector<Error>{ transaction_too_old(), future_version() });
 					}
+					state AnnotateActor annotation(currentLineage);
```

Comment:
Relying on the destructor to determine when to remove the `ActorLineage` from the set is potentially problematic... if the Flow generated state object doesn't get destructed until the end of the actor, the profiler could be taking samples of an actor that is no longer waiting on the original `wait` call but instead same later `wait` call in the actor.